libs/azure-storage/tests/unit_tests/test_document_loaders.py

```diff
@@ -61,9 +61,14 @@ def mock_container_client(
         "langchain_azure_storage.document_loaders.ContainerClient"
     ) as mock_container_client_cls:
         mock_client = MagicMock(spec=ContainerClient)
-        mock_client.list_blob_names.return_value = [
-            blob["blob_name"] for blob in get_test_blobs()
-        ]
+        mock_blobs = []
+        for blob in get_test_blobs():
+            mock_blob = MagicMock()
+            mock_blob.name = blob["blob_name"]
+            mock_blob.size = int(blob["size"])
+            mock_blobs.append(mock_blob)
+
+        mock_client.list_blobs.return_value = mock_blobs
```

Comment:
Would it be possible to add a unit test for both the async and sync version where we ensure that we do not return blobs that match an adls directory? While we have the integration tests, a unit test version is useful as well to capture what the exact heuristic is and also ensure it is always ran in CI since the integration tests are not ran in CI as of now and not as ran as often.